submissions/Dovahkiin1991/friends-app/app.js

```diff
@@ -0,0 +1,339 @@
+                        <h6>${user.dob.age} years old</h6>
+                    </div>
+
+                    <div class="user-description">
+                        <p>${user.location.country + ', ' + user.location.city}</p>
+                        <p>
+                            <a href="mailto:${user.email}">${user.email}</a>
+                        </p>
+                        <p>
+                            <a href="tel:${user.phone}">${user.phone}</a>
+                        </p>
+                        <p class="uppercase">${user.gender}</p>
+                    </div>
+
+                    <div class="user-button" data-id="${user.name.last.toLowerCase()}">
+                        View Profile
+                    </div>
+                </div>
+            </div>
+        </div>
+    `;
+}
+
+const renderUser = (user) => {
+    let singleUser = `
+        <div class="user user--single">
+            <div class="user-block">
+                <div class="user-poster">
+                    <img src="${user.picture.medium}" alt="${user.name.first + ' ' + user.name.last}">
+                </div>
+
+                <div class="user-headings">
+                    <h5><strong>${user.name.first + ' ' + user.name.last}</strong></h5>
+                    <h6>${user.dob.age} years old</h6>
+                </div>
+
+                <div class="user-description">
+                    <p><strong>User Location:</strong></p>
+                    <p>${new Date(user.dob.date).toLocaleDateString('en-us', { weekday:"long", year:"numeric", month:"short", day:"numeric"}) }</p>
+                    <p>&nbsp;</p>
+                    <p><strong>User Location:</strong></p>
+                    <p>${user.location.country}</p>
+                    <p>${user.location.state}</p>
+                    <p>${user.location.city}</p>
+                    <p>&nbsp;</p>
+                    <p><strong>User Contacts:</strong></p>
+                    <p>
+                        <a href="mailto:${user.email}">${user.email}</a>
+                    </p>
+                    <p>
+                        <a href="tel:${user.phone}">${user.phone}</a>
+                    </p>
+                    <p>&nbsp;</p>
+                    <p><strong>User Gender:</strong></p>
+                    <p class="uppercase">${user.gender}</p>
+                </div>
+
+                <div class="user-button user-button--close">
+                    Back to Friends
+                </div>
+            </div>
+        </div>
+    `;
+    mainContainer.innerHTML = singleUser;
+}
+
+// filters
+let filter = document.querySelector('.filter__block input');
+nameInput.addEventListener('input', (e) => {
+    const value = e.preventDefault();
+    const pipes = [filterByName];
+    
+    if (getOpt('alphabet')) {
+        pipes.push(sortByAlphabet);
+    }
+    if (getOpt('age')) {
+        pipes.push(sortByAge);
+    }
+    if (getOpt('gender')) {
+        pipes.push(filterByGender);
+    }
```

Comment:
This approach is pretty strange. You're working not with data, but with HTML state of the page. And, you're also reaching out to DOM multiple times every time someone writes a letter in the search input. While you could've just selected these html elements once and just checked their status.